lib/trino-cache/src/main/java/io/trino/cache/EvictableCacheBuilder.java

```diff
@@ -240,14 +240,12 @@ private static <K, V> CacheLoader<K, V> unimplementedCacheLoader()
     }
 
     @ElementTypesAreNonnullByDefault
-    private static final class TokenWeigher<K, V>
+    private record TokenWeigher<K, V>(Weigher<? super K, ? super V> delegate)
```

Comment:
I don't think we want this. `delegate` becomes part of the public interface of TokenWeighter, where it should be a private detail not accessible to callers.

Reply:
Agreed in general, except this class is private, so it can be used only thru the interface it implements